### PR TITLE
Measures to prevent unassigned activity

### PR DESCRIPTION
**Pull request by zjmiller**

### CODE REVIEW

File: client/src/components/BlockEditor/BlockEditorEditing.tsx [hunk 1]
```diff
@@ -264,18 +264,35 @@ export class BlockEditorEditingPresentational extends React.Component<
...
           }}
         >
+          <div
+            style={{
+              backgroundColor: "#333",
+              bottom: 0,
+              display:
+                this.props.mutationStatus.status === MutationStatus.Error
+                  ? "block"
+                  : "none",
```
[the remark this answers is not in the document]
Comment (zjmiller) [reply client/src/components/BlockEditor/BlockEditorEditing.tsx]: Yeah much better idea. Will switch to this.

File: server/lib/scheduler/Scheduler.ts [hunk 2]
```diff
@@ -170,6 +170,13 @@ class Scheduler {
     const assignedWorkspace = pickRandomItemFromArray(actionableWorkspaces);
     const isThisAssignmentTimed = await assignedWorkspace.hasTimeBudgetOfRootParent;
 
+    // This is intended to mitigate a race condition where two users
+    // are both looking for a new assignment at the same time and
+    // get assigned to the same workspace due to async code in the scheduler
+    if (this.schedule.isWorkspaceCurrentlyBeingWorkedOn(assignedWorkspace)) {
+      return await this.assignNextWorkspace(userId, maybeSuboptimal);
```
Comment (andrewschreiber) [inline server/lib/scheduler/Scheduler.ts]: I see this function throws on line 167 if there are no eligible workspaces, which will prevent infinite loops if all workspaces are taken. Looks good.

**Comment by brachbach** [comment]
@zjmiller anything stopping us from merging this right now?

**Comment by zjmiller** [comment]
Want to test it on staging briefly. Will merge tomorrow morning.

**Pull request merged**